terraform/development/shared/main.tf

```diff
@@ -0,0 +1,168 @@
+
+  required_providers {
+    aws = {
+      version = "~>5.0"
+      source  = "hashicorp/aws"
+    }
+  }
+
+  backend "s3" {
+    bucket         = "core-non-prod-tf-state"
+    dynamodb_table = "core-non-prod-tf-state-lock"
+    encrypt        = true
+    key            = "core-development.tfstate"
+    region         = "eu-west-2"
+    role_arn       = "arn:aws:iam::815624722760:role/developer"
+  }
+}
+
+provider "aws" {
+  region = "eu-west-2"
+
+  assume_role {
+    role_arn = local.provider_role_arn
+  }
+}
+
+provider "aws" {
+  alias  = "us-east-1"
+  region = "us-east-1"
+
+  assume_role {
+    role_arn = local.provider_role_arn
+  }
+}
+
+locals {
+  prefix = "core-dev"
+
+  app_host                  = "review.submit-social-housing-data.levellingup.gov.uk"
+  load_balancer_domain_name = "review.lb.submit-social-housing-data.levellingup.gov.uk"
+
+  provider_role_arn = "arn:aws:iam::837698168072:role/developer"
+
+  enable_aws_shield = false
+
+  application_port = 8080
+  database_port    = 5432
+  redis_port       = 6379
+}
+
+module "application_roles" {
+  source = "../../modules/application_roles"
+
+  github_actions_role_arn = "arn:aws:iam::815624722760:role/core-application-repo"
+
+  prefix                               = local.prefix
+  bulk_upload_bucket_access_policy_arn = module.bulk_upload.read_write_policy_arn
+  database_data_access_policy_arn      = module.database.rds_data_access_policy_arn
+  export_bucket_access_policy_arn      = module.cds_export.read_write_policy_arn
+
+  secret_arns = [
+    module.application_secrets.api_key_secret_arn,
+    module.application_secrets.govuk_notify_api_key_secret_arn,
+    module.application_secrets.os_data_key_secret_arn,
+    module.application_secrets.rails_master_key_secret_arn,
+    module.application_secrets.sentry_dsn_secret_arn
+  ]
+}
+
+module "application_secrets" {
+  source = "../../modules/application_secrets"
+}
+
+module "application_security_group" {
+  source = "../../modules/application_security_group"
+
+  prefix                          = local.prefix
+  application_port                = local.application_port
+  database_port                   = local.database_port
+  db_security_group_id            = module.database.rds_security_group_id
+  load_balancer_security_group_id = module.front_door.load_balancer_security_group_id
+  redis_port                      = local.redis_port
+  vpc_id                          = module.networking.vpc_id
+}
+
+module "bulk_upload" {
+  source = "../../modules/bulk_upload"
+
+  prefix = local.prefix
+}
+
+module "cds_export" {
+  source = "../../modules/cds_export"
+
+  prefix = local.prefix
+}
+
+module "certificates" {
+  source = "../../modules/certificates"
+
+  providers = {
+    aws.us-east-1 = aws.us-east-1
+  }
+
+  cloudfront_domain_name    = local.app_host
+  load_balancer_domain_name = local.load_balancer_domain_name
+}
+
+module "database" {
+  source = "../../modules/rds"
+
+  allocated_storage       = 100

```

Comment:
In PaaS we allocated a DB instance with 5Gb of stage per review app.

In our case we will one DB instance with several DB's inside per review app, so 100Gb felt reasonable to accommodate multiple (~20) review apps at any one time

Reply:
It occurs to me - have you had a look at the actual amount of data typically stored in review apps (or the free space metrics)? If that's a generous amount we may not need 20x the amount we've been giving a single app.

Reply:
On average review apps tend to use ~2.5Gb (half of the storage space available to them, though some breathing space for storage is good, typically at least 10%)

PaaS currently has 22 DBs, but there's only 9 PRs open for the app

**So I'll set this to 50Gb here** (~16/17 apps, each taking 2.5Gb + 20% breathing space => 3Gb each)

---

If we want to be safe we could also configure `max_allocated_storage` in dev, enables autoscaling of the DB storage (without downtime) for the following conditions:
- Free available space is <= 10% of the allocated storage
- The low-storage condition lasts at least 5mins
- At least 6hrs have passed since the last storage modification, or storage optimization has completed on the instance, whichever is longer

The additional storage is in increments of whichever of the following is greater:
- 10 Gb
- 10% of currently allocated storage
- Predicted storage growth exceeding the current allocated storage size in the next 7hrs based on the FreeStorageSpace metrics from the past hour (I'm worried that a burst of review apps over a short time period might unnecessarily trigger this though, it probably works better for gradually reducing free storage space behaviour)

Reply:
Allocated storage now 50Gb. No autoscaling but can update if you think it's necessary 👍